shared/ui-components/src/helpers/useOutsideClick.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useCallback, RefObject } from "react"
+
+type OutsideClickProps = {
+  ref: RefObject<Element>
+  callback: () => void
```

Comment:
Nitpicky, but I'd use `() => unknown` for the type here (since it's not that you require the handling function not to return anything, you just don't care about the return value)

Reply:
Correct, changed to unknown.